### PR TITLE
parameterize MultiPartServletTest for complianceModes and add extra test

### PR DESCRIPTION
**Pull request by lachlan-roberts**
No behaviour changes just improving the test.

- add test to check that maximum number of parts is allowed.
- parameterise tests over RFC7578 and LEGACY compliance modes.

### CODE REVIEW

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java [hunk 1]
```diff
@@ -200,21 +217,57 @@ public void testLargePart() throws Exception
...
+    public void testManyParts(MultiPartFormDataCompliance compliance) throws Exception
+    {
+        connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration()
+            .setMultiPartFormDataCompliance(compliance);
+
+        byte[] byteArray = new byte[10];
+        Arrays.fill(byteArray, (byte)1);
+
+        MultiPartRequestContent multiPart = new MultiPartRequestContent();
+        for (int i = 0; i < 1000; i++)
```
Comment (sbordet) [inline jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java]: IIUC, this is the default value of `maxFormKeys`. If so, make the `ServletContextHandler` in `start()` a field, and here call `contextHandler.getMaxFormKeys()` so this is not a "magic" number.

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java [hunk 2]
```diff
@@ -200,21 +217,57 @@ public void testLargePart() throws Exception
...
-        byte[] byteArray = new byte[1024];
+        connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration()
+            .setMultiPartFormDataCompliance(compliance);
+
+        byte[] byteArray = new byte[5];
         Arrays.fill(byteArray, (byte)1);
 
         MultiPartRequestContent multiPart = new MultiPartRequestContent();
-        for (int i = 0; i < 1024 * 1024; i++)
+        for (int i = 0; i < 1001; i++)
```
Comment (sbordet) [inline jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java]: Ditto.
Number `1001` seems a typo, so use `contextHandler.getMaxFormKeys() + 1` instead.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java (hunk 1 above)
Comment (lachlan-roberts) [reply jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java]: Yes, but I also wanted to test that the default value was reasonable as the typical usage of Jetty doesn't have someone explicitly setting `maxFormKeys`.

Comment (sbordet) [reply jetty-servlet/src/test/java/org/eclipse/jetty/servlet/MultiPartServletTest.java]: Sure, you can assert that separately, something like `context.getMaxFormKeys() > 500 && < 1500`.

**Review by sbordet — APPROVED** [review]

**Review by joakime — APPROVED** [review]

**Comment by joakime** [comment]
@lachlan-roberts you can merge this one now
